### DISCUSSION TITLE
tp.system.prompt for multiple values (tp.system.form?)

### DISCUSSION THREAD

**Discussion by P4cster** [SilentVoid13/Templater#976] [Ideas & Feature Requests]
I was wondering if its possible to implement a form like the system.prompt command. Something like 

`tp.system.form(
form-title: string,
var-name1: string,
var-name1 description: string,
var-name1 default_value?: string|function (tp.system.suggester() as dropdown or something),
var-name2?,...)`

After submitting you should be able to call the user value with `<% var-name# %>` throughout the file. That way there wouldn't be so many prompts one after another. 

Best regards.

**Comment by AB1908** [comment]
I've been thinking about this as well but don't have the time to implement it. Would you like to try?

**Reply by P4cster** [reply]
If I were capable of doing so I wouldn't wonder if its possible :D I just translated my idea into the given templater syntax. Sorry if that misleaded your thoughts.

But happy to see, that I'm not the only one who would welcome such a feature.

Maybe there is some good samaritan out there who could tackle the task?

**Comment by NotNotQuinn** [comment]
I think this would be a great feature! A related discussion is located here: #1059

**Comment by Zachatoo** [comment]
This plugin appears to be tackling this.

https://github.com/danielo515/obsidian-modal-form
